common/src/test/java/io/netty/util/internal/logging/MessageFormatterTest.java

```diff
@@ -0,0 +1,321 @@
+/*
+ * Copyright 2017 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+/**
+ * Copyright (c) 2004-2011 QOS.ch
```

Comment:
@normanmaurer NOTICE.txt already contains lines about slf4j:

> This product contains a modified portion of 'SLF4J', a simple logging
> facade for Java, which can be obtained at:
> 
>   * LICENSE:
>     * license/LICENSE.slf4j.txt (MIT License)
>   * HOMEPAGE:
>     * http://www.slf4j.org/

I just did it by example of `MessageFormatter.java`.